server/public/model/client4.go

```diff
@@ -9205,3 +9205,30 @@ func (c *Client4) SubmitClientMetrics(ctx context.Context, report *PerformanceRe
 
 	return BuildResponse(res), nil
 }
+
+func (c *Client4) GetFilteredUsersStats(ctx context.Context, options *UserCountOptions) (*UsersStats, *Response, error) {
+	query := fmt.Sprintf("?in_team=%v&in_channel=%v&include_deleted=%v&include_bots=%v&include_remote_users=%v",
+		options.TeamId, options.ChannelId, options.IncludeDeleted, options.IncludeBotAccounts, options.IncludeRemoteUsers)
+
+	if len(options.Roles) > 0 {
+		query += fmt.Sprintf("&roles=%s", strings.Join(options.Roles, ","))
+	}
+	if len(options.ChannelRoles) > 0 {
+		query += fmt.Sprintf("&channel_roles=%s", strings.Join(options.ChannelRoles, ","))
+	}
+	if len(options.TeamRoles) > 0 {
+		query += fmt.Sprintf("&team_roles=%s", strings.Join(options.TeamRoles, ","))
+	}
```

Comment:
I think it would be best to handle this with `url.Values` so that there won't be trivial errors for url escaping etc. A simple usage would be:
```go
...
v := url.Values{}
if len(options.Roles) > 0 {
	v.Set("roles", v.Set("since", fmt.Sprintf("%d", options.Roles)))
}
...
```